### PR TITLE
solana-install: Initial commands implementation

### LINKED ISSUE #3222: Design/implement upgrade simple software update mechanism for validator nodes
Validator nodes currently need additional (or manual) infrastructure to update their software.  For example the scripts under the `net/` directory.  However a 3rd party validator node wouldn't have access to the `net/` infrastructure (assuming they don't want to share SSH access to their machine) and are left to their own devices for when to upgrade their validator software.

Let's build a simple updater with roughly the following features:
* [x] An agent running on the machine that periodically polls for software updates.  The machine administrator opts in to running it.
* [x] An on-chain program that tracks the latest software version for each [release channel](https://github.com/solana-labs/solana/blob/master/RELEASE.md#channels).  Each release binary should be be signed by the creator, with the signature and list of download URLs to the release stored on-chain.  Multiple machine architectures should be supported in the design, but initially only Ubuntu Linux binaries are required to be supplied
* [x] No facility to support timing an update (ie, update after block height X).  The upgrades are assumed to be applicable immediately.  
* [x] Incompatible network upgrades (in the short term) can be managed by changing the program id of the on-chain upgrader program.  This will effectively halt all automated upgrades for existing nodes
* [x] The updater agent itself should of course be updated.
* [x] (Potentially) consider a rollback mechanism, as well as a mechanism for nodes to report update success.

### PR DESCRIPTION
**Pull request by mvines**
Fill in implementation of the `solana-install` commands:

TODO:
* [x] Complete the `run` command implementation
* [x] Beautify console output
* [x] Show progress bar while downloading an update
* [ ] Add some tests

Fixes #3222

### CODE REVIEW

**Comment by mvines** [comment]
Here's an example of how a new cluster software update can be deployed by somebody:
https://asciinema.org/a/oYEaSxTDlpTxYrKAqWY5ioCIt

And here's the client side of this: a validator that wants to receive software updates:
https://asciinema.org/a/FbA2TfDc8UCcLmpxYLsQkTC7B

**Comment by mvines** [comment]
@CriesofCarrots - hey would you have some time to review this?  If not just decline, np

**Comment by CriesofCarrots** [comment]
@mvines, yep! Can you give me 20min?

**Comment by mvines** [comment]
> @mvines, yep! Can you give me 20min?

I'd be ecstatic with 20 hours :)

**Comment by CriesofCarrots** [comment]
@mvines, great! I will wait a bit, then, so I don’t have to review on my phone :)

File: proposals/src/installer.md [hunk 1]
```diff
@@ -52,12 +50,12 @@ $ solana-keygen ...  # <-- runs the latest solana-keygen
 $ solana-install run solana-fullnode ...  # <-- runs a fullnode, restarting it as necesary when an update is applied
 ```
 
-### Update Manifest on-chain program
-The Update Manifest program is used to advertise the deployment of new release tarballs
-on a solana cluster.  Each instance of this program describes a logical update
-channel for a given target triple (eg, `x86_64-apple-darwin`).  The public key
-of each program instance is well-known between the entity deploying new updates
-and users consuming those updates.
+### on-chain update manifest
```
Comment (CriesofCarrots) [inline proposals/src/installer.md]: Capitalize this head

**Pull request merged**